common/types/encoding/da.go

```diff
@@ -42,16 +66,32 @@ func (w *Block) NumL1Messages(totalL1MessagePoppedBefore uint64) uint64 {
 }
 
 // NumL2Transactions returns the number of L2 transactions in this block.
-func (w *Block) NumL2Transactions() uint64 {
+func (b *Block) NumL2Transactions() uint64 {
 	var count uint64
-	for _, txData := range w.Transactions {
+	for _, txData := range b.Transactions {
 		if txData.Type != types.L1MessageTxType {
 			count++
 		}
 	}
 	return count
 }
 
+// EstimateL1CommitCalldataSize calculates the calldata size in l1 commit approximately.
+// TODO: The calculation could be more accurate by using 58 + len(l2TxDataBytes) (see Chunk).
+// This needs to be adjusted in the future.
+func (b *Block) EstimateL1CommitCalldataSize() uint64 {
```

Comment:
This belongs to `v0`, since `v1` will have a different calldata size.

Reply:
nice catch, will change this and review other parts.